src/tools/pkcs11-tool.c

```diff
@@ -2683,7 +2683,11 @@ derive_key(CK_SLOT_ID slot, CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key)
 		const EC_GROUP *ecgroup = NULL;
 		const EC_POINT * ecpoint = NULL;
 		
-		buf = malloc(buf_size);	/* TODO: caculate some sensible value from key length */	
+		buf_size = 512; /* TODO: calculate buffer size from key length */	
+		buf = malloc(buf_size);	
+		
+		if (!buf)
+		    util_fatal("Failed to allocate memory for other party's public point.")
```

Comment:
please update your pull request by squashing the clean up commit onto the actual commit that contain the real logic